src/index.js

```diff
@@ -33,9 +34,12 @@ function polymorphicMapper(iterableSchema, itemSchema, bag, options) {
   };
 }
 
-function visitIterable(obj, iterableSchema, bag, options) {
-  const itemSchema = iterableSchema.getItemSchema();
-  const curriedItemMapper = defaultMapper(iterableSchema, itemSchema, bag, options);
+function visitIterable(obj, iterableSchema, parentId, bag, options) {
+  let itemSchema = iterableSchema.getItemSchema();
+  if(iterableSchema._mappedBy) {
+    itemSchema = clone(itemSchema).mappedBy(iterableSchema._mappedBy);
```

Comment:
Do we need a *deep* clone here? Does this library need to depend on a very complex `clone()` implementation for the sake of this line? Wouldn’t a shallow clone be enough here?

It seems like if we don’t make `mappedBy` mutating in the first place, we can drop dependency on `clone()`.